### ISSUE TITLE
PHP7 should be supported

### ISSUE THREAD

**Issue by petrkotek** [braintree/braintree_php#149]
Hi,

People will be more & more interested in PHP7 support (I believe there's no need to mention the reasons/benefits) and the `README.md` only states that "PHP5.4+" is supported, however since there are some incompatibilities in PHP7, it would be nice to state the support explicitely.

So.. would it be possible to update us about plans for PHP7 support  here?
- I can see that the tests are passing under PHP7 (see https://github.com/petrkotek/braintree_php/pull/1)
- Are there currently any known issues with PHP7?
- Are there any plans around official support of PHP7?
- Is there some timeline?
- Can community help somehow? (for the moment I've submitted at least #148 to run unit tests in all supported PHP versions for each PR/push/merge)

Thanks in advance for your response!

**Comment by jackellenberger** [comment]
Hi petrkotek,

Would you mind clarifying what incompatibilities you expect to see with PHP7? We aren't regularly testing against that version, but we will be adding it to our internal CI at your suggestion. Should we run into any issues we'll make them known and roll out a patch in the next version. We don't have an official timeline as of now, but we do mean "compatibility with PHP5.4+", so PHP7 support will be forthcoming.

**Comment by petrkotek** [comment]
Thanks Jack.

### TL;DR
I investigated it a bit more & there shouldn't be any issues with the code -- i.e. everything should work under PHP7 as expected. It would however be nice to "support it officially". 

### Long answer:

I took a look into that a bit more & ran [`php7cc`](https://github.com/sstalle/php7cc), which tries to identify potential issues (I've got good experience with it).

It doesn't detect any issues for `braintree/braintree_php` repo, which is great:
```
braintree_php $ git rev-parse HEAD
d6ce80f7f54abe9972d577173343d1975bc49e77
braintree_php $ php7cc -x vendor/ .
Checked 210 files in 2.956 seconds
```

Since you don't even have any dependency on other libraries, it's looking very good.

PHPUnit 3.7 is quite old and not supported on PHP7, however it works & it's only dev dependency, so that's not a concern.

It would however be nice to "support it officially".

**Comment by jackellenberger** [comment]
Hi @petrkotek,

Thanks for your explanation, and again for bringing this to our attention. We're currently adding more up to date versions of PHP to our CI suite, and we'll be updating our language on version support to be a bit more explicit shortly.

**Comment by petrkotek** [comment]
Thanks a lot, highly appreciated 👍
